### PR TITLE
[22.01] Fix tool-util packaging

### PR DESCRIPTION
**Pull request by mvdbeek**
Fixes https://github.com/galaxyproject/galaxy/issues/14527

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: packages/tool_util/HISTORY.rst [hunk 1]
```diff
@@ -5,6 +5,17 @@ History
...
+22.1.2 (2022-08-29)
+-------------------
+
+* Fix lint context error level
+* Pin galaxy-util to >= 22.1
+
+22.1.1 (2022-08-22)
+--------------------
+
+* First release from the 22.01 branch of Galaxy
```
Comment (mr-c) [inline packages/tool_util/HISTORY.rst:18]: Thanks. Any hint on how to where to look for the API breaking changes in 22.01?

https://github.com/common-workflow-language/cwltool/issues/1722

Comment (mvdbeek) [reply packages/tool_util/HISTORY.rst:18]: https://github.com/galaxyproject/galaxy/commit/da61e77068f115ebe0afcd10a6d34b9dd43aaae7 ? That looks broken, no idea what I wanted to do with that `except KeyError`

Comment (mr-c) [reply packages/tool_util/HISTORY.rst:18]: ah.. So cwltool should just pin an earlier version of galaxy-tool-util until this gets fixed?

We also have a problem that the galaxy biocontainer resolver is not pulling from biocontainers, but is instead mulling an entire new container locally. This is very slow!

But that doesn't seem to be a new problem; I'm having difficulty in finding out when that changed (sometime after July 2017)

Comment (mvdbeek) [reply packages/tool_util/HISTORY.rst:18]: That would certainly explain why you ran into this error in the first place. Let me have a look at cwltool

Comment (mvdbeek) [reply packages/tool_util/HISTORY.rst:18]: OK, I took the core of https://github.com/common-workflow-language/cwltool/blob/9eeccee69df2b3f0a70bc414616446bdfcc6c952/cwltool/software_requirements.py#L133-L156 and turned that into a unit test, and now I also understand why I worked with `KeyError`, it's not as crazy as it seems (it's a beaker object) but we should have definitely had type annotations.

Comment (mr-c) [reply packages/tool_util/HISTORY.rst:18]: Cool. Any hints as to why we are getting locally mulled containers instead of a quay.io biocontainer reference?

Comment (mvdbeek) [reply packages/tool_util/HISTORY.rst:18]: Not yet, but the unit test is doing that too, so I'm going to fix that

**Review by jmchilton — APPROVED** [review]

**Comment by mr-c** [comment]
As of 2f150595432aee2ae364c12d63d1e564c611fcb4 this fixes `cwltool --beta-use-biocontainers`; thanks!

A point release would be very appreciated; I think this has been broken for a while

**Comment by mvdbeek** [comment]
Will do once this is in. Given the +1 that'll just be once the tests have passed.

**Review by nsoranzo — APPROVED** [review]
I've added a commit with mainly type annotations, hopefully that looks OK.

File: lib/galaxy/tool_util/deps/container_resolvers/__init__.py [hunk 2]
```diff
@@ -4,7 +4,9 @@
     abstractmethod,
     abstractproperty,
 )
+from typing import Optional
 
+from beaker.cache import Cache
```
Comment (mvdbeek) [inline lib/galaxy/tool_util/deps/container_resolvers/__init__.py]: beaker isn't a core tool-util requirement, so won't that break ? that's why I didn't add this.

Comment (nsoranzo) [reply lib/galaxy/tool_util/deps/container_resolvers/__init__.py]: I'll move it under `if TYPE_CHECKING:`

File: packages/tool_util/HISTORY.rst [hunk 3]
```diff
@@ -5,6 +5,17 @@ History
 
 .. to_doc
 
+22.1.2 (2022-08-29)
+-------------------
+
+* Fix lint context error level
+* Pin galaxy-util to >= 22.1
```
Comment (mr-c) [inline packages/tool_util/HISTORY.rst:12]: Does this need updating now?

File: lib/galaxy/tool_util/deps/container_resolvers/__init__.py [hunk 4]
```diff
@@ -4,11 +4,18 @@
...
+    Optional,
+    TYPE_CHECKING,
+)
 
 
 from galaxy.util.bunch import Bunch
 from galaxy.util.dictifiable import Dictifiable
 
+if TYPE_CHECKING:
+    from beaker.cache import Cache
```
Comment (mvdbeek) [inline lib/galaxy/tool_util/deps/container_resolvers/__init__.py:17]: Won't that still break type checking in e.g cwltool ?

Comment (mr-c) [reply lib/galaxy/tool_util/deps/container_resolvers/__init__.py:17]: We can install https://pypi.org/project/Beaker/ (yes?) for type checking, no problem.

File: packages/tool_util/HISTORY.rst (hunk 3 above)
Comment (mvdbeek) [reply packages/tool_util/HISTORY.rst:12]: Yep, how about 

```suggestion
22.1.2 (2022-08-29)
-------------------

* Fix lint context error level
* Pin galaxy-util to >= 22.1
* Fix biocontainer resolution without beaker cache
```

?

Comment (mr-c) [reply packages/tool_util/HISTORY.rst:12]: LGTM

**Comment by mvdbeek** [comment]
@mr-c I've uploaded https://pypi.org/project/galaxy-tool-util/22.1.2/

**Pull request merged**